chart/scripts/post-delete-hook.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -e
+
+namespaces="${NAMESPACES}"
+timeout="${TIMEOUT}"
+ignoreTimeoutError="${IGNORETIMEOUTERROR}"
+
+if [ -z "${namespaces}" ]; then
+  echo "No namespace is provided."
+  exit 1
+fi
+
+if [ -z "${timeout}" ]; then
+  echo "No timeout value is provided."
+  exit 1
+fi
+
+if [ -z "${ignoreTimeoutError}" ]; then
+  echo "No ignoreTimeoutError value is provided."
+  exit 1
+fi
+
+succeeded=()
+failed=()
+
+get_pod_count() {
+  kubectl get pods --selector app="${1}" -n "${2}" -o json | jq '.items | length'
+}
+
+echo "Uninstalling Rancher resources in the following namespaces: ${namespaces}"
+
+for namespace in ${namespaces}; do
+  for app in $(helm list -n "${namespace}" -q); do
+    if [[ ${app} =~ .crd$ ]]; then
+      echo "--- Skip the app [${app}] in the namespace [${namespace}]"
+      continue
+    fi
+    echo "--- Deleting the app [${app}] in the namespace [${namespace}]"
+    if [[ ! $(helm uninstall "${app}" -n "${namespace}") ]]; then
+      failed=("${failed[@]}" "${app}")
+      continue
+    fi
+    t=0
+    # by default, wait 120 seconds in total for an app to be uninstalled
+    while [[ $(get_pod_count "${app}" "${namespace}") != "0" ]] && [[ "${t}" -lt "${timeout}" ]]; do
```

Comment:
```suggestion
    while [[ "$(get_pod_count ${app} ${namespace})" -ne 0 ]] && [[ "${t}" -lt "${timeout}" ]]; do
```

Reply:
1. The double brackets `[[` prevents the word splitting by default, so the quotes around the string variable are optional. 
I replaced other `[`s with `[[`s in the file and no quotes to make use of the good feature of  `[[`. 

2. `"$(get_pod_count ${app} ${namespace})"` returns a string not a number, so we need to compare two strings here.

Reply:
1. Sounds good! Thanks for the clarification
2. Bash doesn’t have the concept of types so strings or numbers don’t make a difference. A number is a “string” with just digits